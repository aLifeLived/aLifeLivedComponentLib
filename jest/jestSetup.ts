/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
