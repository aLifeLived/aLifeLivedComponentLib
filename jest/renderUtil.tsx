import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
