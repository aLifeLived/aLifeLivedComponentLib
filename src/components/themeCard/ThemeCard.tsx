import React from 'react';
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { SystemImage } from '../../themes/systemImage/SystemImage';

type ThemeCardProps = {
  title: string;
  imageUrl?: string;
  subText: string;
  onPress: () => void;
};

export const ThemeCard: React.FC<ThemeCardProps> = ({
  title,
  imageUrl,
  subText,
  onPress,
}) => {
  return (
    <SystemButton
      onPress={() => onPress()}
      borderWidth={1}
      borderRadius="medium"
      backgroundColor="white"
      borderColor="borderColor"
    >
      <Box boxShadow="md" flexDirection="row">
        {imageUrl ? (
          <SystemImage
            source={{ uri: imageUrl, priority: 'high' }}
            resizeMode="cover"
            width={70}
            height={70}
            borderTopLeftRadius="medium"
            borderBottomLeftRadius="medium"
          />
        ) : null}
        <Box
          padding="s"
          justifyContent="center"
          borderTopRightRadius="medium"
          borderBottomRightRadius="medium"
          width="80%"
        >
          <Text variant="h4" fontWeight="bold" numberOfLines={1}>
            {title}
          </Text>
          <Text variant="bodySmall" numberOfLines={1} marginTop="xs">
            {subText}
          </Text>
        </Box>
      </Box>
    </SystemButton>
  );
};
