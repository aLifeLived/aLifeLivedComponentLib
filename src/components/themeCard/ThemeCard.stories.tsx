import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

// Component
import { ThemeCard } from './ThemeCard';

// UI
import { Box } from '../../themes/box/Box';
import { FlatList } from 'react-native';

const data = [
  {
    id: '1',
    title: 'TEST',
    imageUrl: 'https://picsum.photos/id/237/200/300',
    subText: 'test subtext',
  },
  {
    id: '2',
    title: 'TEST',
    imageUrl: 'https://picsum.photos/id/237/200/300',
    subText: 'test subtext',
  },
  {
    id: '3',
    title: 'TEST',
    imageUrl: 'https://picsum.photos/id/237/200/300',
    subText: 'test subtext',
  },
  {
    id: '4',
    title: 'TEST',
    imageUrl: 'https://picsum.photos/id/237/200/300',
    subText: 'test subtext',
  },
];

storiesOf('Cards', module)
  .addDecorator(Story => Story())
  .add('Default theme card in list', () => (
    <Box flex={1} backgroundColor="white">
      <FlatList
        data={data}
        numColumns={2}
        keyExtractor={item => item.id}
        renderItem={() => (
          <Box margin="m" width="40%">
            <ThemeCard
              title={text('Title', 'A Life Lived')}
              imageUrl="https://picsum.photos/id/237/200/300"
              onPress={action('On press fired')}
            />
          </Box>
        )}
      />
    </Box>
  ));
