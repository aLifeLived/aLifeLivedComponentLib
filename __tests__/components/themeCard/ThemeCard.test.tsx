import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { ThemeCard } from '../../../src/components/themeCard/ThemeCard';

const onPress = jest.fn();
const defaultTitle = 'Test Title';
const defaultImageUrl = 'https://picsum.photos/id/237/200/300';

const setup = ({ imageUrl }: { imageUrl: string }) =>
  render(
    <ThemeCard title={defaultTitle} imageUrl={imageUrl} onPress={onPress} />
  );

describe('components >> theme card ', () => {
  beforeEach(() => onPress.mockClear());

  test('Should match snapshot correctly', () => {
    const { toJSON } = setup({ imageUrl: defaultImageUrl });
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should find title text correctly', () => {
    const { getByText } = setup({ imageUrl: defaultImageUrl });
    expect(getByText(defaultTitle)).toBeDefined();
  });

  test('Should fire on press action', () => {
    const { getByText } = setup({ imageUrl: defaultImageUrl });
    fireEvent.press(getByText(defaultTitle));
    expect(onPress).toBeCalledTimes(1);
  });
});
