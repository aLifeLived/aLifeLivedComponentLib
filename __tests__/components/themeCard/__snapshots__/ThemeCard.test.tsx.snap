// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> theme card  Should match snapshot correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "space-between",
      "backgroundColor": "#F1ECE5",
      "borderRadius": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "marginLeft": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "display": "flex",
              "paddingRight": 16,
              "paddingTop": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#282828",
                "display": "flex",
                "fontFamily": "SFUIText-Regular",
                "fontSize": 20,
                "lineHeight": 24,
              },
            ]
          }
        >
          Test Title
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "borderRadius": 16,
            "display": "flex",
            "overflow": "hidden",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "borderBottomLeftRadius": 8,
                "borderTopLeftRadius": 8,
                "height": 100,
                "width": 100,
              },
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="cover"
          source={
            Object {
              "priority": "high",
              "uri": "https://picsum.photos/id/237/200/300",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
